src/github.com/getlantern/idletiming/idletiming_conn.go

```diff
@@ -91,6 +100,13 @@ func (c *IdleTimingConn) TimesOutAt() time.Time {
 
 // Read implements the method from io.Reader
 func (c *IdleTimingConn) Read(b []byte) (int, error) {
+	c.closeMutex.RLock()
+	defer c.closeMutex.RUnlock()
```

Comment:
Just commenting here, but do we really want to hold on to `closeMutex` for the entire read? That looks like it will make simultaneous reads and writes impossible. Maybe a comment on `closeMutex` specifying what exactly it's protecting (presumably `closed` but if that's the case I would probably put the lock in `checkClosed`)